### PR TITLE
fix: allow application to set and query app version

### PR DESCRIPTION
**Pull request by cmwaters**
## Description

This add the necessary API to fix the bug with restarting the application that was caused by not being able to correctly set the app version

### CODE REVIEW

File: baseapp/abci.go
Comment (rootulp) [inline baseapp/abci.go]: [nit]

```suggestion
// CreateQueryContext creates a new sdk.Context for a query, taking as args
```

File: baseapp/baseapp.go
Comment (rootulp) [inline baseapp/baseapp.go]: [nit] this comment appears residual

```suggestion
```

**Review by evan-forbes — APPROVED** [review]

**Comment by evan-forbes** [comment]
nice find

File: baseapp/baseapp.go
Comment (rootulp) [inline baseapp/baseapp.go:226]: [question] is the removal of this public method _technically_ breaking? I don't think we need to cut a major release of Cosmos SDK as long as know that celestia-app has to replace usage of `InitAppVersion` when it bumps to a new release of Cosmos SDK with this PR.

Side note: Is this change back-portable to v1.x? If it isn't, then we may need to create a new branch in this repo if there's a bug we need to address on celestia-app v1.x after this PR merges. Perhaps not worth worrying about for now.

**Review by rootulp — APPROVED** [review]
LGTM after one residual comment gets removed

File: baseapp/baseapp.go
Comment (cmwaters) [reply baseapp/baseapp.go:226]: I could keep this function around if we were worried about breaking changes. I had intended to replace the usage of `InitAppVersion`.

Comment (rootulp) [reply baseapp/baseapp.go:226]: I'm more concerned with the ability to backport than the breaking change. IMO if it's possible to backport to celestia-app v1.x easily then I'm fine with removing `InitAppVersion`. If it isn't possible to backport then we may have to maintain multiple Cosmos SDK branches (one per celestia-app major version) which seems like it could be a maintenance burden.

File: baseapp/baseapp.go
Comment (rach-id) [inline baseapp/baseapp.go:233]: What is app version 0? Just a non-empty value to return?

File: baseapp/baseapp.go
Comment (cmwaters) [reply baseapp/baseapp.go:226]: It's not used in v1.x so it should be easy to backport

File: baseapp/baseapp.go
Comment (cmwaters) [reply baseapp/baseapp.go:233]: Yeah, it just means that it's not current set

**Review by ninabarbakadze — APPROVED** [review]

**Review by rootulp — APPROVED** [review]

**Pull request merged**
